src/Datadog.Trace.ClrProfiler.Managed/Integrations/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -49,34 +58,58 @@ public static T ExecuteSyncImpl<T>(object multiplexer, object message, object pr
         /// <param name="server">The server to call.</param>
         /// <returns>An asynchronous task.</returns>
         public static object ExecuteAsyncImpl<T>(object multiplexer, object message, object processor, object state, object server)
+        {
+            return ExecuteAsyncImplInternal<T>(multiplexer, message, processor, state, server);
```

Comment:
What is the purpose of forwarding the call to `ExecuteAsyncImplInternal` that takes exactly the same arguments?

Reply:
The "internal" methods use `async`/`await` which require a return type of `Task` or `Task<T>`, but we inject methods with `object` return type. The method signatures must match (including the return type), which is why I split them into 2 methods.